UnitTests/Tests/ProductServiceTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Moq;
+using NUnit.Framework;
+using SportsGoods.App.Services;
+using SportsGoods.Core.Interfaces;
+using SportsGoods.Core.Models;
+using SportsGoods.Data.DAL;
+using System.Reflection;
+
+namespace UnitTests.Tests
+{
+    [TestFixture]
+    public class ProductServiceTests
+    {
+        private ApplicationDbContext? _context = null;
+
+        [SetUp]
+        public async Task Setup()
+        {
+            var connectionString = "Server=(localdb)\\MSSQLLocalDB;Database=SportsGoodsTest;Trusted_Connection=True;TrustServerCertificate=True;";
+
+            var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+                .UseSqlServer(connectionString)
+            .Options;
+
+            _context = new ApplicationDbContext(options);
+
+            _context.Products.RemoveRange(_context.Products);
+            await _context.SaveChangesAsync();
+        }
+
+        [Test]
+        public async Task SeedProductsFromXml_ValidXml_AddsProductsToDatabase()
+        {
+            var solutionDirectory = GetSolutionDirectory();
+            var testDataDirectory = Path.Combine(solutionDirectory, "SolutionItems");
+            var xmlFilePath = Path.Combine(testDataDirectory, "products.xml");
+
+            var productService = new ProductService(_context);
+
+            var initialProductCount = _context.Products.Count();
+
+            await productService.SeedProductsFromXmlAsync(xmlFilePath);
+
+            var finalProductCount = _context.Products.Count();
+            Assert.That(finalProductCount, Is.GreaterThan(initialProductCount));
+        }
+
+        [Test]
+        public async Task SeedProductsFromXml_InvalidXml_DoesNotAddProductsToDatabase()
+        {
+            var xmlFileName = "invalid_products.xml";
+            var testDataDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..", "..", "..", "TestData");
+            var xmlFilePath = Path.Combine(testDataDirectory, xmlFileName);
+
+            var productService = new ProductService(_context);
+
+            var initialProductCount = _context.Products.Count();
+
+            await productService.SeedProductsFromXmlAsync(xmlFilePath);
+
+            var finalProductCount = _context.Products.Count();
+            Assert.That(initialProductCount, Is.EqualTo(finalProductCount));
+        }
+
+        [Test]
+        public void SeedProductsFromXml_NonExistentXmlFile_ThrowsFileNotFoundException()
+        {
+            var nonExistentXmlFilePath = "nonexistent.xml";
+
+            var productService = new ProductService(_context);
+
+            Assert.Throws<FileNotFoundException>(() =>
```

Comment:
I believe you can use https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.ThrowsAsync.html

Reply:
Done (refactored)